File: Singleton/Singleton.php
<?php
/**
 * Created by PhpStorm.
 * User: jamesyan
 * Date: 2017/3/14
 * Time: 22:18
 */

class Singleton{
    private static $instance;

    private function __construct()
    {

    }

    public static function getInstance(){

        if(!self::$instance instanceof self){
            self::$instance=new self();
            echo "11111<br/>";
        }
        echo  "22222<br/>";
        return self::$instance;
    }

    public function showMessage(){
        echo "hello world<br/>";
    }

}

//test

$object=Singleton::getInstance();
$object->showMessage();

$object2=Singleton::getInstance();
$object2->showMessage();

File: Singleton/Singleton.go
package main

import "fmt"

var _instance *Singleton

type Singleton struct {
	name string
}

func getInstance() *Singleton {
	if _instance == nil {
		_instance = new(Singleton)
		fmt.Println(11111)
	}
	fmt.Println(22222);
	return _instance
}

func (this *Singleton) setName(name string) {
	this.name = name;
}

func (this *Singleton) showMessage() {
	fmt.Println("hello world", this.name);
}

func main() {
	object := getInstance();
	object.setName("object1")
	object.showMessage();
	object2 := getInstance();
	object2.setName("object2")
	object2.showMessage();
}